torch/nn/quantized/modules/conv.py

```diff
@@ -155,7 +164,8 @@ def get_qconv(cls, mod, activation_post_process, weight_post_process=None):
         assert weight_post_process.dtype == torch.qint8, \
             'Weight observer must have a dtype of qint8'
         qweight = _quantize_weight(mod.weight.float(), weight_post_process)
-        qconv = cls(mod.in_channels, mod.out_channels, mod.kernel_size,
+        # the __init__ call used is the one from derived classes and not the one from _ConvNd
+        qconv = cls(mod.in_channels, mod.out_channels, mod.kernel_size,  # type: ignore[call-arg]
```

Comment:
So the issue is that the subclass `__init__` methods have two fewer parameters (no `transposed, output_padding`) than `_ConvNd`. The current fix works, but technically the current way the class hierarchy is organized is not correct and the better way to do it would be:

```
    def __init__(self, in_channels, out_channels, kernel_size, stride=1,
                 padding=0, dilation=1, groups=1, bias=True,
                 padding_mode='zeros'):
        # All subclasses have this signature
        raise NotImplementedError

    def _init(self, in_channels, out_channels, kernel_size, stride,
              padding, dilation,
              transposed, output_padding,
              groups, bias,
              padding_mode='zeros'):
        super(_ConvNd, self).__init__()
```
and then subclasses should call `super(Conv1d, self)._init(`

I think that would be a good change, but it is only possible if there are no subclasses of `_ConvNd` floating around (it's private, but you never know).

Let's see what other reviewers think.

Reply:
Ok, I'll wait the other reviewers before changing this part of the code

Reply:
Soooo it is a little tricky. `torch.nn._ConvNd` is sadly de facto public API; see https://github.com/pytorch/fairseq/blob/master/fairseq/modules/quantization/scalar/modules/qconv.py#L14 for an example. But this is *quantized* ConvNd, and this code is relatively new and there ought to be less uses. cc @jerryzh168 for more thoughts.

I'm trying to decide if I agree that the alternate version is better.

Reply:
I guess the high level question I have is why the subclass init methods have fewer parameters. This seems like a blatant oversight to me.

Reply:
> I guess the high level question I have is why the subclass init methods have fewer parameters

Because those two parameters are implicitly defined by the dimensionality of subclass - e.g. `Conv1d` hardcodes `transpose, output_padding` to `False, (0,)`. It would not make sense for the user to define those when initializing `Conv1d`.

`transpose` is always `False`, so that really doesn't make much sense - probably code just copied from the non-quantized version or something like that.

Reply:
OK fair enough. The suggested alternative seems like a reasonable way to shut up the type checker then.

Reply:
I've updated the code to use `_init`.

Reply:
I see, are we going to have similar changes in https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/conv.py as well?